sklearn/inspection/_plot/partial_dependence.py

```diff
@@ -878,6 +927,42 @@ def plot(self, *, ax=None, n_cols=3, line_kw=None, contour_kw=None):
         import matplotlib.pyplot as plt  # noqa
         from matplotlib.gridspec import GridSpecFromSubplotSpec  # noqa
 
+        # FIXME: remove in 1.2
+        if self.pdp_lim != "deprecated":
+            if pdp_lim is not None and self.pdp_lim != pdp_lim:
+                warnings.warn(
+                    "`pdp_lim` has been passed in both the constructor and the `plot` "
+                    "method. For backward compatibility, the parameter from the "
+                    "constructor will be used. To silence this warning, only use "
+                    "`pdp_lim` from the `plot` method.",
+                    UserWarning,
+                )
+            pdp_lim = self.pdp_lim
+            warnings.warn(
+                "The `pdp_lim` parameter is deprecated in version 1.0 and will be "
+                "removed in version 1.2. Provide `pdp_lim` to the `plot` method."
+                "instead.",
+                FutureWarning,
+            )
```

Comment:
Probably switching warning makes more sense when both are raised? Also, the now-proposed second warning remark (regarding providing `pdp_lim` to `plot`) is redundant.
```suggestion
            warnings.warn(
                "The `pdp_lim` parameter is deprecated in version 1.0 and will be "
                "removed in version 1.2. Provide `pdp_lim` to the `plot` method."
                "instead.",
                FutureWarning,
            )
            if pdp_lim is not None and self.pdp_lim != pdp_lim:
                warnings.warn(
                    "`pdp_lim` has been passed in both the constructor and the `plot` "
                    "method. For backward compatibility, the parameter from the "
                    "constructor will be used.",
                    UserWarning,
                )
            pdp_lim = self.pdp_lim
```